### PR TITLE
Fixes #4842 feat: Add the API Endpoints for Favourite Events of a User

### LINKED ISSUE #4842: Add the API Endpoints for Favourite Events of a User
**Is your feature request related to a problem? Please describe.**
The general app for Events will utilise a feature where Users will be able to to favourite an Event and there will be a section too where they can see their favourited events.

**Describe the solution you'd like**
Add the API Endpoints to Post favourite events and Get favourited events of a user

**Additional context**
Working on it : No

### PR DESCRIPTION
**Pull request by bhaveshAn**
Fixes #4842 feat: Add the API Endpoints for Favourite Events of a User

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

Adds the API Endpoints for Favourite Events of a User

#### Changes proposed in this pull request:

- Adds the API Endpoints for Favourite Events of a User

screenshot from 2018-08-01 20-04-43
screenshot from 2018-08-01 20-05-02
screenshot from 2018-08-01 20-05-43

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -224,7 +228,7 @@
           '/attendees/<int:attendee_id>/event', '/custom-forms/<int:custom_form_id>/event',
           '/orders/<order_identifier>/event', '/faqs/<int:faq_id>/event', '/faq-types/<int:faq_type_id>/event',
           '/feedbacks/<int:feedback_id>/event', '/stripe-authorizations/<int:stripe_authorization_id>/event',
-          '/event-chat-messages/<int:message_id>/event')
+          '/event-chat-messages/<int:message_id>/event', '/user-favourite-events/<int:user_favourite_event_id>/event')
```
Comment (Kreijstal) [inline app/api/__init__.py]: There is no chat system

Comment (bhaveshAn) [reply app/api/__init__.py]: Done. Removed.

File: app/models/user_favourite_event.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from app.models import db
+
+
+class UserFavouriteEvent(db.Model):
```
Comment (srv-twry) [inline app/models/user_favourite_event.py]: Soft deletion since it is user related.

File: app/api/schema/user_favourite_events.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                         schema='EventSchemaPublic',
+                         type_='event')
+    user = Relationship(attribute='user',
+                        self_view='v1.user_favourite_event_user',
+                        self_view_kwargs={'id': '<id>'},
+                        related_view='v1.user_detail',
+                        related_view_kwargs={'user_favourite_event_id': '<id>'},
+                        schema='UserSchema',
+                        type_='user')
+    deleted_at = fields.DateTime(dump_only=True)
```
Comment (srv-twry) [inline app/api/schema/user_favourite_events.py]: You don't need to add this field if you are already overriding the soft deletion schema.

**Pull request merged**